### PR TITLE
feat : added delete endpoint

### PR DESCRIPTION
**Pull request by BichraiX**
Added this endpoint from the spec : https://spec.matrix.org/v1.11/client-server-api/#post_matrixclientv3account3piddelete

### CODE REVIEW

File: packages/matrix-client-server/src/account/3pid/3pid.test.ts [hunk 1]
```diff
@@ -412,16 +707,17 @@ describe('Use configuration file', () => {
           ok: true,
           status: 200,
           // eslint-disable-next-line @typescript-eslint/promise-function-async
-          json: () =>
-            Promise.resolve({
+          json: () => {
+            return {
```
Comment (guimard) [inline packages/matrix-client-server/src/account/3pid/3pid.test.ts:711]: I don't understand this change: the real API doesn't return a promise ?

File: packages/matrix-client-server/src/account/3pid/delete.ts [hunk 2]
```diff
@@ -16,15 +23,126 @@ interface RequestBody {
...
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(openIDResponseBody)
+    }
+  )
+  const validToken = (
+    registerResponse.json() as unknown as RegisterResponseBody
```
Comment (guimard) [inline packages/matrix-client-server/src/account/3pid/delete.ts:93]: Should be `(await registerResponse.json()) as unknown as RegisterResponseBody` if I'm right in previous comment

File: packages/matrix-client-server/src/account/3pid/delete.ts [hunk 3]
```diff
@@ -34,48 +152,93 @@ const delete3pid = (clientServer: MatrixClientServer): expressAppHandler => {
...
+              clientServer.matrixDb
+                .get('user_threepid_id_server', ['id_server'], {
+                  user_id: data.sub,
+                  medium: body.medium,
+                  address: body.address
+                })
+                .then((rows) => {
+                  if (rows.length === 0) {
+                    clientServer.logger.error(
+                      'No id_server found corresponding to the user'
```
Comment (guimard) [inline packages/matrix-client-server/src/account/3pid/delete.ts:212]: Could you replace "user" by the name given by token ? It's better to set the name of user when available in  400 responses (bad request)

File: packages/matrix-client-server/src/account/3pid/delete.ts (hunk 2 above)
Comment (BichraiX) [reply packages/matrix-client-server/src/account/3pid/delete.ts:93]: Yes you're right I've fixed it thank you !

File: packages/matrix-client-server/src/account/3pid/delete.ts (hunk 3 above)
Comment (BichraiX) [reply packages/matrix-client-server/src/account/3pid/delete.ts:212]: Yes done

File: packages/matrix-client-server/src/account/3pid/3pid.test.ts (hunk 1 above)
Comment (BichraiX) [reply packages/matrix-client-server/src/account/3pid/3pid.test.ts:711]: It does but in all mocks in identity service and in the mock right before this one there was no Promise.resolve for a similar API call, so I decided to remove it to match the rest of the code

Comment (guimard) [reply packages/matrix-client-server/src/account/3pid/3pid.test.ts:711]: We need to reproduce exactly the express API everywhere in test, so if `req.json()` returns a promise, mock must do the same everywhere

**Pull request merged**